### ISSUE TITLE
Don't persist state - allow apps to handle state

### ISSUE THREAD

**Issue by larrysalibra** [stx-labs/stacks.js#265]
We should stop storing stuff in `localStorage` in our javascript library. See https://forum.blockstack.org/t/duplication-of-code-state-in-blockstack/1539?u=larry

See the problems this causes for web workers #264

**Comment by larrysalibra** [comment]
Today we ran into a developer issue where his development environment already had some data stored in `localStorage.get('blockstack')` from earlier testing. This behavior was making him think that the manual way he was using to process the blockstack authentication process was working locally but not in production when in reality it wasn't working locally, but the `loadUserData()` method was reading stale leftover data.

**Comment by timkinnane** [comment]
I am that developer 👍 @larrysalibra I also created a new mostly empty Blockstack user profile today and found that `.loadUserData()` was still returning my original profile, while logged in as the new one. I could only get the new users details by emptying local storage and setting up the keychain again.

**Comment by timkinnane** [comment]
@larrysalibra I encountered the problem I mentioned in my last comment again today, this time after using the 'Reset Browser' option from Blockstack settings. This is a pretty big security issue. If a user resets their browser, then somebody else uses the browser, they can still access the prior user's profile. In the case where the new user also uses Blockstack auth, they might actually inherit the former's attributes if they sign into an app. Should I make another issue for this, or is it the same thing?

**Comment by icidasset** [comment]
Any update on this? I want to bring Blockstack into V2 of my app, but I’m blocked by this 😥 Need to use Blockstack in a web worker.

**Comment by friedger** [comment]
You can define your own store since 19.1

**Comment by icidasset** [comment]
@friedger Awesome thanks!
